hail/src/main/scala/is/hail/expr/ir/functions/StringFunctions.scala

```diff
@@ -82,10 +85,83 @@ object StringFunctions extends RegistryFunctions {
 
   def escapeString(s: String): String = StringEscapeUtils.escapeString(s)
 
+  def splitQuoted(s: String, separator: String, missing: Array[String], quote: String): Array[String] = {
+    if (quote.size != 1 && quote != null) throw new HailException(s"quote length cannot be greater than 1," +
+                                                 s" quote length entered ${quote.size}")
+
+    val quoteC = if (quote == null) '\u0000' else quote.charAt(0)
+
+    val ab = new BoxedArrayBuilder[String]()
+    val sb = new StringBuilder()
+
+    val testIfMissing = (field_entry: String) => {
+      if (!missing.contains(field_entry)) field_entry
+      else null
+    }
+    val matchSep: Int => Int = separator.length match {
```

Comment:
I think this is adding a lot of overhead. For each line you'll need to allocate this anonymous class, and the matchSep becomes a virtual function call on that class. 

However, I see this doesn't differ between implementations. we should fix this in the future.